File: src/AutoRest.SdkExplorer/Model/Schema/SchemaMethodParameter.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using AutoRest.SdkExplorer.Model.Code;

namespace AutoRest.SdkExplorer.Model.Schema
{
    public class SchemaMethodParameter

    {
        public string? Name { get; set; }
        public string? RelatedPropertySerializerPath { get; set; }
        public TypeDesc? Type { get; set; }
        public bool IsOptional { get; set; }
        public string? DefaultValue { get; set; }
        public string Description { get; set; } = string.Empty;

        public SchemaMethodParameter()
        {

        }
    }
}

File: src/AutoRest.SdkExplorer.TypeScript/src/Model/Schema/SchemaMethodParameter.ts
import { TypeDesc } from "../Code/TypeDesc";

export class SchemaMethodParameter implements AutoRest.SdkExplorer.Interface.SchemaMethodParameter {
    name: string;
    relatedPropertySerializerPath: string;
    type: TypeDesc;
    isOptional: boolean;
    defaultValue?: string;
    description: string;

    constructor(data: AutoRest.SdkExplorer.Interface.SchemaMethodParameter) {
        this.name = data.name!;
        this.relatedPropertySerializerPath = data.relatedPropertySerializerPath!;
        this.type = new TypeDesc(data.type!);
        this.isOptional = data.isOptional!;
        this.defaultValue = data.defaultValue;
        this.description = data.description ?? "";
    }
}
